Add make variable for setting python 2 variable for cpplint

#### Summary:
cpplint requires Python2. Add a variable to set it from the command line.

#### Reproducible Steps:
When `python` defaults to Python 3, cpplint fails to run.

#### Current Output:
```
> make cpplint RUN_CPPLINT="python3 lib/cpplint_4.45/cpplint.py" 
make: *** [cpplint] Error 1
```

#### Current Version:
v2.9.0

--- a/make/cpplint
+++ b/make/cpplint
@@ -1,5 +1,5 @@
-RUN_CPPLINT ?= python $(CPPLINT)/cpplint.py
+PYTHON2 ?= python
 
 .PHONY: cpplint
 cpplint:
-	@$(RUN_CPPLINT) --output=vs7 --counting=detailed --root=stan --extension=hpp,cpp --filter=-runtime/indentation_namespace,-readability/namespace,-legal/copyright,-whitespace/indent,-runtime/reference $(shell find stan -name '*.hpp' -o -name '*.cpp')
+	@$(PYTHON2) $(CPPLINT)/cpplint.py --output=vs7 --counting=detailed --root=stan --extension=hpp,cpp --filter=-runtime/indentation_namespace,-readability/namespace,-legal/copyright,-whitespace/indent,-runtime/reference $(shell find stan -name '*.hpp' -o -name '*.cpp')

--- a/makefile
+++ b/makefile
@@ -121,6 +121,8 @@ endif
 	@echo '  - cpplint       : runs cpplint.py on source files. requires python 2.7.'
 	@echo '                    cpplint is called using the CPPLINT variable:'
 	@echo '                      CPPLINT = $(CPPLINT)'
+	@echo '                    To set the version of python 2, set the PYTHON2 variable:'
+	@echo '                      PYTHON2 = $(PYTHON2)'
 	@echo ''
 	@echo 'Documentation:'
 	@echo '  Doxygen'

